### PR TITLE
feat(stream-client): implement delete stream

### PR DESCRIPTION
**Pull request by ppaidi**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: TBD
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
Delete stream from designer page is not working

**What is the new behavior?**
Implemented delete stream from designer page

### CODE REVIEW

File: libs/plugins/stream-client/src/lib/core/stream.service.ts [hunk 1]
```diff
@@ -137,4 +144,42 @@ export class StreamService {
...
+      .pipe(
+        switchMap(translation => {
+          return this.confirmationModal.openModal({
+            title: translation['MODAL:CONFIRM-DELETION'],
+            textMessage: translation['STREAMS.DELETE-STREAM:CONFIRM_DELETE'],
+          }).result;
+        }),
+        filter(result => result === ConfirmationResult.Confirm),
+        switchMap(() => {
+          return this.appResourceService.deleteResourceWithTrigger(stream.id, null).pipe(
```
Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/core/stream.service.ts]: We can update the logic such that we do not have to have an empty subscription in the component. 
```
this.translate
      .get()
      .pipe(
        switchMap(translation => {
          return this.confirmationModal.openModal().result;
        }),
        filter(result => result === confirm),
        switchMap(() => this.appResourceService.deleteResourceWithTrigger(stream.id, null))
      ).subscribe(() => {
          this.navigateToApp(stream.app.id);
          notify user
        }, err => {
          console.error(err);
          this.notifications.error({
            key: 'STREAMS.DELETE-STREAM:ERROR-MESSAGE-DELETE-STREAM',
            params: err,
          });
        });
```

Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/core/stream.service.ts]: Why are we always calling the `deleteResourceWithTrigger` with `null` as second parameter?

File: libs/plugins/stream-client/src/lib/stream-designer/stream-designer.component.ts [hunk 2]
```diff
@@ -62,7 +62,13 @@ export class StreamDesignerComponent implements OnInit, OnDestroy {
     this.isStreamMenuOpen = !this.isStreamMenuOpen;
   }
 
-  deleteStream() {}
+  deleteStream() {
+    this.closeStreamMenu();
+    this.streamService
+      .deleteStream(this.streamState)
+      .pipe(takeUntil(this.ngOnDestroy$))
```
Comment (apandurangi) [inline libs/plugins/stream-client/src/lib/stream-designer/stream-designer.component.ts]: if we change the definition of `deleteStream` we do not need to subscribe here.

**Comment by ppaidi** [comment]
Delete unreferenced triggers when a resource is deleted is tracked in #1153

**Pull request merged**